pkg/scraper/component_scraper.go

```diff
@@ -160,18 +161,42 @@ func getResourceHealthStatus(ctx context.Context, k8sClient ctrclient.Client, ob
 		return nil, err
 	}
 
-	if health.Status == common.HealthStatusProgressing {
-		unhealthyPods, err := common.HasUnhealthyPods(ctx, k8sClient, obj)
-		if err != nil {
-			return nil, err
-		}
-		if unhealthyPods {
-			return &common.HealthStatus{
-				Status: common.HealthStatusDegraded,
-			}, nil
+	if obj.GetAnnotations() == nil {
```

Comment:
can we refactor extracting the last found timestamp into a separate function (and maybe add the timestamp back in a defer block if not found?).  That will make it more clean to read and reusable.